franka_hardware/src/franka_hardware_interface.cpp

```diff
@@ -133,7 +133,13 @@ CallbackReturn FrankaHardwareInterface::on_activate(
 CallbackReturn FrankaHardwareInterface::on_deactivate(
     const rclcpp_lifecycle::State& /*previous_state*/) {
   RCLCPP_INFO(getLogger(), "trying to Stop...");
-  robot_->stopRobot();
+  try {
+    robot_->stopRobot();
+  } catch (const franka::Exception& e) {
+    RCLCPP_ERROR(getLogger(), e.what());
+    return CallbackReturn::ERROR;
```

Comment:
I guess when there is an error in this stage. HardwareInterface will go back to the on_configure() stage. Is it possible that the robot initializes itself after error is acknowledged. Or what exactly will happen when we are back in the on_configure() stage?

Reply:
@BarisYazici Thank you, sorry for the late reply. I am not sure how or when the robot initializes itself. Does that happen in firmware, or in `libfranka`, or both? I do not know much about Franka internals or `libfranka`, so I am probably not best equipped to write code which manages Franka's internal state.

Reply:
@BarisYazici Thank you, sorry for the late reply. I am not sure how or when the robot initializes itself. Does that happen in firmware, or in `libfranka`, or both? I do not know much about Franka internals or `libfranka`, so I am probably not best equipped to write code which manages Franka's internal state.

Reply:
It was a question more on how you want to use this feature. Because if the robot goes in the reflex mode, by throwing an exception, the hardware interface will not be functioning. But the controllers will still be running without noticing that the hardware interface is in Error state. 

It's mostly a question of how you wanna use this feature and how the ROS 2 hardware interface work. So with this change the hardware interface and the robot will not be updating but the controllers will continue to work. I don't think this is an expected behavior for the users. So unless we have a logic to recover the robot from the error state, I don't see the value of this feature.  So maybe you can mention the use-case you have and we can try to suit the feature to your use-case. Maybe this involves going back to the configure stage or introducing a recovery state in the hardware interface. But in this current stage if we catch the exception we will not allow the exception to escalate, which might be misleading because they wouldn't understand why their controller stopped working.